core/commonTest/src/contract/map/ImmutableMapTest.kt

```diff
@@ -283,13 +283,16 @@ abstract class ImmutableMapTest {
     }
 
     @Test fun noOperation() {
-        immutableMapOf<Int, String>().toPersistentMap().testNoOperation({ clear() }, { clear() })
+        immutableMapOf<Any, String>().toPersistentMap().testNoOperation({ clear() }, { clear() })
 
-        val map = immutableMapOf("x" to 1, null to "x").toPersistentMap()
+        val instance = Any()
+        val instance2 = Any()
+
+        val map = immutableMapOf("x" to instance, null to "x").toPersistentMap()
         with(map) {
             testNoOperation({ remove("y") }, { remove("y") })
-            testNoOperation({ remove("x", 2) }, { remove("x", 2) })
-            testNoOperation({ put("x", 1) }, { put("x", 1) })     // does not hold
+            testNoOperation({ remove("x", instance2) }, { remove("x", instance2) })
```

Comment:
Weird: The `remove` provided for builder, is actually called on `map`.

Reply:
Makes sense to eliminate the use of `with` scope function

Reply:
I would suggest the following changes to the test: https://github.com/Kotlin/kotlinx.collections.immutable/commit/f3e3c61058be51937fc5e8c922ab93587607abb6
WDYT?